### ISSUE TITLE
Feature Idea: Add `roots` to config

### ISSUE THREAD

**Issue by supergibbs** [terraform-linters/tflint#950]
There are a couple related issues (#527, #841) but want to start a separate thread to get feedback on this idea.

It seems common to have terraform repos with multiple folders/projects within it. A `--recursive` option would help but since it would lose context, only some rules could be checked and modules wouldn't work. You can write a script to traverse and `tflint` each but then everyone needs to maintain their own. To better support this structure, I propose a new key in the `config` section called `roots` which would be an array of paths to all the folders you want checked. This would also allow a shared config file so you can have a common set of rules for all your projects. If it supported globbing, you could come up with a convention like `roots = ["oneoff/config", "**/src",]`. 

Somewhat related, it would be nice if the `.tflint.hcl` priority was:
```
./.tflint.hcl (local)
../.tflint.hcl (recursively search for .tflint.hcl in parent folders)
~/.tflint.hcl (fallback on home dir)
```
This would allow overriding config in needed but allow a default at the repo level. Bonus points if the local `.tflint.hcl` could have a flag `override = true` which would still search for parent/home .tflint.hcl and merge/override that.

An alternative would be create a `root` object that could optionally have its own config and rules that would override the top level defaults. It could also support globbing.

```
root {
  path = "oneoff/config"
  config { .... }
  rule "some_rule" { enabled = false }
}

root {
  path = "**/src"
}
```

Then running `tflint` would run in each root specified. Maybe even in parallel?

**Comment by bendrucker** [comment]
I put some time into the walking up for configuration idea and merging idea:

https://github.com/terraform-linters/tflint/tree/walk-up-config

Definitely interested in that. 

Less inclined to make TFLint itself find modules recursively. But upward configuration lookup would simplify existing user scripts and add the ability to override configuration on a per-module basis.

**Comment by wata727** [comment]
I've been thinking about this issue for a long time, but haven't been able to get a good answer.

First of all, as you said, the problem with the `--recursive` option is that the context is lost. This context includes the Terraform's one. For example, Terraform behaves differently due to various factors such as environment variables (`TF_VAR_*`, `TF_DATA_DIR`, `TF_WORKSPACE`... etc.), the current directory, and the Terraform's workspace.

Most of these are difficult to follow even with extended TFLint config file. With that in mind, I think it's best to implement a user script that simulates the Terraform context at this time...

**Comment by supergibbs** [comment]
@bendrucker Awesome, I think a walk up is a great first step and would make a homegrown run script or manual run easier in multi project repos.

@wata727 That is a good point, if env vars are relied on, that can make things more complicated. 

The more complex option of a `root {}` object could solve that by defining env vars and anything else there but then we're getting out of scope. Maybe something like that should be a separate tool.

Thanks for the feedback guys!
